### PR TITLE
Update ORTModule API docstrings

### PR DESCRIPTION
**Pull request by natke**
**Description**: Update the API docs for ORTModule

**Motivation and Context**
- So that we can generate the API docs for ORTModule

### CODE REVIEW

**Review by baijumeswani — DISMISSED** [review]

File: orttraining/orttraining/python/training/ortmodule/ortmodule.py [hunk 1]
```diff
@@ -52,8 +52,24 @@ def _forward(self, *inputs, **kwargs):
...
-        ...
+        '''Delegate the :meth:`~torch.nn.Module.forward` pass of PyTorch training to
+        ONNX Runtime.
+
+        The first call to forward performs setup and checking steps. During this call,
+        ORTModule determines whether the module can be trained with ONNX Runtime. For
+        this reason, the first forward call execution takes longer than subsequent calls.
+        Execution is interupted if ONNX Runtime cannot process the model for training.
+
+        args:
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/ortmodule/ortmodule.py:63]:
```suggestion
        Args:
```
NIT

File: orttraining/orttraining/python/training/ortmodule/ortmodule.py [hunk 2]
```diff
@@ -52,8 +52,24 @@ def _forward(self, *inputs, **kwargs):
...
+            *inputs and **kwargs represent the positional, variable positional, keyword
+            and variable keyword arguments defined in the user's PyTorch module's forward
+            method. Values can be torch tensors and primitive types.
+
+        returns:
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/ortmodule/ortmodule.py:68]:
```suggestion
        Returns:
```
NIT

**Review by thiagocrepaldi — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

File: orttraining/orttraining/python/training/ortmodule/ortmodule.py [hunk 3]
```diff
@@ -52,8 +52,24 @@ def _forward(self, *inputs, **kwargs):
...
+        Args:
+            *inputs and **kwargs represent the positional, variable positional, keyword
+            and variable keyword arguments defined in the user's PyTorch module's forward
+            method. Values can be torch tensors and primitive types.
+
+        Returns:
+            The output as expected from the forward method defined by the user's
+            PyTorch module. Output values supported include tensors, nested sequences
+            of tensors and nested dictionaries of tensor values.
+        '''
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/ortmodule/ortmodule.py:72]: Missing a Raises block for the torch cpp extension scenario

```
Raises:
    ORTModuleInitException: When PyTorch C++ Extensions for ORTModule is not installed. users must execute `python -m torch_ort.configure`

**Pull request merged**